packages/react/src/vertex4.tsx

```diff
@@ -0,0 +1,322 @@
+    textboxStrokeWidth = 1,
+    textFontFamily = "Verdana",
+    annotationGutter = 2,
+    annotations = [],
+    iconAnnotations = [],
+    cornerRadius = 3,
+    icon = {},
+    subText = {},
+    showLabel = true,
+    noLabelRadius = 5,
+
+    iconBorderWidth = 1,
+    iconBorderColor = "#333",
+
+    iconBackgroundColor = "#fff",
+
+    iconFontColor = "#000",
+    iconFontSize=20,
+    iconFontFamily="FontAwesome",
+
+    shapeOffsetX = 0,
+    shapeOffsetY = 0,
+    iconOffsetX = 0,
+    iconOffsetY = 0,
+
+    iconPadding = 4,
+    iconText="?",
+    shapeRendering="auto"
+}) => {
+    icon = {
+        height: 50,
+        imageChar: "?",
+        imageFontFamily: "FontAwesome",
+        imageCharFill: "#555555",
+        fill: "transparent",
+        strokeWidth: 0,
+        ...icon
+    };
+    subText = {
+        text: "",
+        fill: "white",
+        textFill: "#555555",
+        ...subText
+    };
+
+    const annoOffsetY = 0;
+    const labelWidth = React.useMemo(() => {
+        return Utility.textSize(text, textFontFamily, textHeight, false).width;
+    }, [text, textFontFamily, textHeight]);
+
+    let labelShapeWidth = 0;
+    if (text !== "") {
+        labelShapeWidth = labelWidth + (textPadding * 2) + (textboxStrokeWidth * 2);
+    }
+    let fullAnnotationWidth = labelShapeWidth + annotationGutter;
+    const textOffsetX = fullAnnotationWidth - (labelShapeWidth / 2);
+    
+    const textShapeHeight = textHeight + (annotationGutter*2) + (textboxStrokeWidth * 2);
+    const annoWidthArr = React.useMemo(() => {
+        return annotations.map((anno, i) => {
+            return Utility.textSize(anno.imageChar, anno.imageFontFamily, anno.height, false).width;
+        });
+    }, [annotations]);
+    const annotationArr = React.useMemo(() => {
+        const annotationArr = [];
+        let _labelAnnoOffsetX = fullAnnotationWidth;
+        annotations.forEach((anno, i) => {
+            const annoText = anno.imageChar;
+            const annoTextHeight = anno.height ?? textShapeHeight;
+            _labelAnnoOffsetX += annoWidthArr[i] + annotationGutter;
+            const annoOffsetX = _labelAnnoOffsetX - (annoWidthArr[i] / 2);
+            annotationArr.push(
+                <g key={i} class="vertex3-anno" data-anno={JSON.stringify(anno)} transform={`translate(${annoOffsetX} ${annoOffsetY})`}>
+                    <Icon
+                        {...anno}
+                        shape="rectangle"
+                        width={annoWidthArr[i]}
+                        height={annoTextHeight}
+                        imageChar={annoText}
+                        imageFontFamily={anno.imageFontFamily}
+                        cornerRadius={cornerRadius}
+                        strokeWidth={0}
+                    />
+                </g>
+            );
+        });
+        return annotationArr;
+    }, [fullAnnotationWidth, annotations, textShapeHeight, annoWidthArr, annotationGutter, cornerRadius]);
```

Comment:
This will only update if the annotations array reference changes, not if the content of the annotations array changes.